src/EfficientDynamoDb/Internal/Operations/BatchExecuteStatement/BatchExecuteStatementResponseParser.cs

```diff
@@ -0,0 +1,28 @@
+﻿using EfficientDynamoDb.DocumentModel;
+using EfficientDynamoDb.Internal.Operations.Shared;
+using EfficientDynamoDb.Operations.BatchExecuteStatement;
+using System;
+using System.Collections.Generic;
+
+namespace EfficientDynamoDb.Internal.Operations.BatchExecuteStatement
+{
+    internal static class BatchExecuteStatementResponseParser
+    {
+        public static BatchExecuteStatementResponse Parse(Document response)
+        {
+            var responsesArray = response.TryGetValue("Responses", out var responsesAttribute)
+                ? (IReadOnlyList<AttributeValue>)responsesAttribute.AsListAttribute().Items
+                : Array.Empty<AttributeValue>();
+
+            var items = new Document[responsesArray.Count];
+            for (var i = 0; i < responsesArray.Count; i++)
+                items[i] = responsesArray[i].AsDocument()["Item"].AsDocument();
```

Comment:
Batch response parsing doesn't account for some edge cases:
1. If there is no item with provided primary key in the table, the response document won't contain the `Item` property (only `TableName` is there).
2. If operation within batch fails to any reason, the response document won't contain the `Item` property either. There will be an `Error` object instead with error code and message. The `Error` object can also contain `Item` inside, if PartiQL query had `RETURNING ALL OLD *` specified (or other `RETURNING` options).
3. Some operations in batch can fail while other being successful. This case should be gracefully handled because throwing a single exception will be misleading.

Taking in account that it's low level API I'd suggest simply returning `responsesArray[i].AsDocument()` and let clients deal with their particular scenarios. I'm afraid that attempt of handling them in library will result in similar response API that is already returned by DynamoDB. So there is not much sense in changing it in the first place. 

However, I'm open to better ideas if you have any.

Reply:
I thought a bit more about it and I think it's better to have a fully mapped response class after the [BatchStatementResponse](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BatchStatementResponse.html) instead of returning `Document`. So that only `Item` and `Error.Item` properties will have `Document` type. This way the API will be much easier to use even with `Item` and `Error` properties being nullable.

Reply:
Having a fully mapped response class after BatchStatementResponse would look like this:

```csharp
public class BatchExecuteStatementResponse
{
    public IReadOnlyList<FullConsumedCapacity> ConsumedCapacity { get; set; } = null!;

    public IReadOnlyList<BatchStatementResponse> Responses { get; set; } = null!;
}

public class BatchStatementResponse
{
    private Document Error { get; set; } = null!;

    private Dictionary<string, Document> Item { get; set; } = null!;

    private string TableName { get; set; } = null!;
}
```

Is this okay?

Reply:
- All 3 properties in `BatchStatementResponse` should be nullable because there are cases when API returns null. 
- It's possible to map `Error` property as well.
- `Item` is just a `Document` because Document under the hood is `Dictionary<string, AttributeValue>`

So it will look like this:

```csharp
public class BatchStatementResponse
{
    public BatchStatementError? Error { get; set; }

    public Document? Item { get; set; }

    public string? TableName { get; set; }
}

public class BatchStatementError
{
    public string? Code { get; set; }

    public string? Message { get; set; }

    public Document? Item { get; set; }
}
```

Reply:
Okay, I understand